python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -0,0 +1,741 @@
+        dtype_to_pylibcudf_type(binop_out_ty),
+        dtype_to_pylibcudf_type(binop_lhs_ty),
+        dtype_to_pylibcudf_type(binop_rhs_ty),
+        binary_operator,
+    ):
+        fail = True
+    return (binop_lhs_ty, binop_rhs_ty, binop_out_ty, fail)
+
+
+def _test_binaryop_inner(test, columns, pyop, cuop):
+    binop_lhs_ty, binop_rhs_ty, binop_out_ty, fail = test
+    lhs = columns[binop_lhs_ty]
+    rhs = columns[binop_rhs_ty]
+    pylibcudf_outty = dtype_to_pylibcudf_type(binop_out_ty)
+    if not fail:
+        expect_data = pyop(
+            plc.interop.to_arrow(lhs).to_numpy(),
+            plc.interop.to_arrow(rhs).to_numpy(),
+        ).astype(binop_out_ty)
+        expect = pa.array(expect_data)
+        got = plc.binaryop.binary_operation(lhs, rhs, cuop, pylibcudf_outty)
+        assert_column_eq(got, expect)
+    else:
+        with pytest.raises(TypeError):
+            plc.binaryop.binary_operation(lhs, rhs, cuop, pylibcudf_outty)
+
+
+@pytest.fixture(scope="module")
+def add_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.ADD,
+    )
+
+
+@pytest.fixture(scope="module")
+def sub_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.SUB,
+    )
+
+
+@pytest.fixture(scope="module")
+def mul_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.MUL,
+    )
+
+
+@pytest.fixture(scope="module")
+def div_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.DIV,
+    )
+
+
+@pytest.fixture(scope="module")
+def true_div_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.TRUE_DIV,
+    )
+
+
+@pytest.fixture(scope="module")
+def floor_div_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.FLOOR_DIV,
+    )
+
+
+@pytest.fixture(scope="module")
+def mod_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.MOD,
+    )
+
+
+@pytest.fixture(scope="module")
+def pmod_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.PMOD,
+    )
+
+
+@pytest.fixture(scope="module")
+def pymod_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    # TODO
```

Comment:
I suspect that we should only test pymod and not even expose the other mod function. I'm not completely sold on that yet though, just my first impression since I don't know if we want to expose a non-Python mod definition from libcudf to Python.